postgis/gserialized_gist_2d.c

```diff
@@ -1302,6 +1304,182 @@ Datum gserialized_gist_same_2d(PG_FUNCTION_ARGS)
 	PG_RETURN_POINTER(result);
 }
 
+PG_FUNCTION_INFO_V1(gserialized_gist_sortsupport_2d);
+/*
+ * Hash comparator of `GSERIALIZED` data type
+*/
+static int
+hash_cmp(Datum a, Datum b, SortSupport ssup)
+{
+	if (a > b)
+		return 1;
+	else if (a < b)
+		return -1;
+	else
+		return 0;
+}
+
+static bool
+hash_abbrev_abort(int memtupcount, SortSupport ssup)
+{
+	return false;
+}
+
+/*
+ * `BOX2DF`(geometry) data type converter
+*/
+
+inline static uint64_t
+uint64_interleave_2(uint64_t x, uint64_t y)
+{
+	x = (x | (x << 16)) & 0x0000FFFF0000FFFFULL;
+	x = (x | (x << 8)) & 0x00FF00FF00FF00FFULL;
+	x = (x | (x << 4)) & 0x0F0F0F0F0F0F0F0FULL;
+	x = (x | (x << 2)) & 0x3333333333333333ULL;
+	x = (x | (x << 1)) & 0x5555555555555555ULL;
+
+	y = (y | (y << 16)) & 0x0000FFFF0000FFFFULL;
+	y = (y | (y << 8)) & 0x00FF00FF00FF00FFULL;
+	y = (y | (y << 4)) & 0x0F0F0F0F0F0F0F0FULL;
+	y = (y | (y << 2)) & 0x3333333333333333ULL;
+	y = (y | (y << 1)) & 0x5555555555555555ULL;
+
+	return x | (y << 1);
+}
+
+/* Based on https://github.com/rawrunprotected/hilbert_curves Public Domain code */
+inline static uint64_t
```

Comment:
TODO: make sure we have only one copy of this in code

Reply:
I would like to know if I could make the hash function in `gbox.c` visible to other mudule

Reply:
you'll have to remove "static" qualifier for that. alternatively there's "lwinline" thingy where you can move it and include to both places.